extensions/railway/src/railway-projects.tsx

```diff
@@ -0,0 +1,222 @@
+  Detail,
+  environment,
+  List,
+  OpenInBrowserAction,
+  popToRoot,
+  PushAction,
+  showHUD,
+} from "@raycast/api";
+import fs from "fs";
+import mkdirp from "mkdirp";
+import path from "path";
+import { useEffect, useMemo, useState } from "react";
+import {
+  consumeLoginSession,
+  createLoginSession,
+  fetchProjects,
+  fetchUser,
+  logout,
+  ProjectGQL,
+  projectUrl,
+  railwayWebUrl,
+  UserGQL,
+} from "./railway";
+
+const railwayConfigPath = path.resolve(environment.supportPath, "config.json");
+
+const saveLoginToken = (token: string) => {
+  mkdirp.sync(environment.supportPath);
+
+  const contents = JSON.stringify({ token });
+  fs.writeFileSync(railwayConfigPath, contents, "utf8");
+};
+
+const deleteLoginToken = () => {
+  fs.writeFileSync(railwayConfigPath, JSON.stringify({}), "utf8");
+};
+
+const useRailwayToken = (): string | null =>
+  useMemo(() => {
+    try {
+      const { token } = JSON.parse(fs.readFileSync(railwayConfigPath, "utf8"));
+      return token;
+    } catch (e) {}
+
+    return null;
+  }, [environment.supportPath]);
+
+const useUser = () => {
+  const token = useRailwayToken();
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<UserGQL | null>(null);
+
+  useEffect(() => {
+    if (token == null) {
+      setIsLoading(false);
+      return;
+    }
+
+    (async () => {
+      const user = await fetchUser(token);
+      setUser(user);
+      setIsLoading(false);
+    })();
+  }, []);
+
+  return { token, isLoading, user };
+};
+
+export default function Command() {
+  const { user, isLoading, token } = useUser();
+
+  if (isLoading) {
+    return <List isLoading />;
+  }
+
+  if (user == null || token == null) {
+    return <RailwayLogin />;
+  }
+
+  return <ListProjects token={token} />;
+}
+
+const ListProjects: React.FC<{ token: string }> = ({ token }) => {
+  const [projects, setProjects] = useState<ProjectGQL[] | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      const projects = await fetchProjects(token);
+      setProjects(projects);
+    })();
+  }, []);
+
+  return (
+    <List isLoading={projects == null}>
+      {projects != null && (
+        <>
+          {projects.map((p) => (
+            <List.Item
+              key={p.id}
+              title={p.name}
+              subtitle={p.description}
+              actions={
+                <ActionPanel>
+                  <ActionPanel.Section>
+                    <OpenInBrowserAction title="Project settings" url={projectUrl(p.id, "settings")} />
+                    <OpenInBrowserAction title="Project deployments" url={projectUrl(p.id, "deployments")} />
+                    <OpenInBrowserAction title="Latest deployment" url={projectUrl(p.id, "deployments?open=true")} />
+                    <OpenInBrowserAction title="Project variables" url={projectUrl(p.id, "variables")} />
+                    <OpenInBrowserAction title="Project metrics" url={projectUrl(p.id, "metrics")} />
```

Comment:
```suggestion
                    <OpenInBrowserAction title="Open Settings" url={projectUrl(p.id, "settings")} />
                    <OpenInBrowserAction title="Open Deployments" url={projectUrl(p.id, "deployments")} />
                    <OpenInBrowserAction title="Open Latest Deployment" url={projectUrl(p.id, "deployments?open=true")} />
                    <OpenInBrowserAction title="Open Variables" url={projectUrl(p.id, "variables")} />
                    <OpenInBrowserAction title="Open Metrics" url={projectUrl(p.id, "metrics")} />
```

Reply:
Just made all action names title case. Would it also make sense to add another section with some `CopyToClipboardAction`? Could be handy to quickly share stuff...

BTW you can also assign [keyboard shortcuts](https://developers.raycast.com/api-reference/user-interface/action-panel#actionpanel.item) to actions, e.g. `⌘` `L` could open the latest deployment.